client/navigation/utils.ts

```diff
@@ -171,42 +250,64 @@ export const sortMenuItems = ( menuItems ) => {
  * @return {Object} Mapped menu items and categories.
  */
 export const getMappedItemsCategories = (
-	menuItems,
-	currentUserCan = null
-) => {
-	const categories = { ...defaultCategories };
-
-	const items = sortMenuItems( menuItems ).reduce( ( acc, item ) => {
-		// Set up the category if it doesn't yet exist.
-		if ( ! acc[ item.parent ] ) {
-			acc[ item.parent ] = {};
-			menuIds.forEach( ( menuId ) => {
-				acc[ item.parent ][ menuId ] = [];
-			} );
-		}
+	menuItems: Item[],
+	currentUserCan: ( capability: string ) => boolean
+): {
+	items: {
+		[ key: string ]: Category | { [ key: string ]: Item[] };
```

Comment:
This could be a lot less wordy as `Record<string, Category> | Record<string, Item[]>`